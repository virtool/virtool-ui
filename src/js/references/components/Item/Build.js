import React from "react";
import { Link } from "react-router-dom";
import { RelativeTime } from "../../../base";
import { ReferenceItemInfo } from "./Info";

export const ReferenceItemBuild = ({ id, latestBuild, progress }) => {
    if (latestBuild && progress === 100) {
        return (
            <ReferenceItemInfo>
                <h4>
                    Latest Build is{" "}
                    <Link to={`/refs/${id}/indexes/${latestBuild.id}`}>Index {latestBuild.version}</Link>
                </h4>
                <p>
                    Created <RelativeTime time={latestBuild.created_at} /> by {latestBuild.user.id}
                </p>
            </ReferenceItemInfo>
        );
    }

    const to = {
        pathname: `/refs/${id}/indexes`,
        state: {
            rebuild: true
        }
    };

    return (
        <ReferenceItemInfo>
            <h4>
                No index found. <Link to={to}>Build one. </Link>
            </h4>
            <p>You cannot use this reference until you have built an index for it.</p>
        </ReferenceItemInfo>
    );
};
