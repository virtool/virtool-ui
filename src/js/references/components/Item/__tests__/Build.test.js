import { ReferenceItemBuild } from "../Build";

describe("<ReferenceItemBuild />", () => {
    let props;

    beforeEach(() => {
        props = {
            id: "foo",
            latestBuild: {
                created_at: "2018-01-01T00:00:00.000000Z",
                id: "bar",
                user: {
                    id: "bob"
                },
                version: 3
            },
            progress: 100
        };
    });

    it("should render build info when latestBuild available and [progress=100]", () => {
        const wrapper = shallow(<ReferenceItemBuild {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render 'No Build Found' when [latestBuild=null]", () => {
        props.latestBuild = null;
        const wrapper = shallow(<ReferenceItemBuild {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render 'No Build Found' when [progress=30]", () => {
        props.progress = 30;
        const wrapper = shallow(<ReferenceItemBuild {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
