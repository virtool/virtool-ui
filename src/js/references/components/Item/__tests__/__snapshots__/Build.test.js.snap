// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReferenceItemBuild /> should render 'No Build Found' when [latestBuild=null] 1`] = `
<Info__ReferenceItemInfo>
  <h4>
    No index found. 
    <Link
      to={
        Object {
          "pathname": "/refs/foo/indexes",
          "state": Object {
            "rebuild": true,
          },
        }
      }
    >
      Build one. 
    </Link>
  </h4>
  <p>
    You cannot use this reference until you have built an index for it.
  </p>
</Info__ReferenceItemInfo>
`;

exports[`<ReferenceItemBuild /> should render 'No Build Found' when [progress=30] 1`] = `
<Info__ReferenceItemInfo>
  <h4>
    No index found. 
    <Link
      to={
        Object {
          "pathname": "/refs/foo/indexes",
          "state": Object {
            "rebuild": true,
          },
        }
      }
    >
      Build one. 
    </Link>
  </h4>
  <p>
    You cannot use this reference until you have built an index for it.
  </p>
</Info__ReferenceItemInfo>
`;

exports[`<ReferenceItemBuild /> should render build info when latestBuild available and [progress=100] 1`] = `
<Info__ReferenceItemInfo>
  <h4>
    Latest Build is
     
    <Link
      to="/refs/foo/indexes/bar"
    >
      Index 
      3
    </Link>
  </h4>
  <p>
    Created 
    <RelativeTime
      time="2018-01-01T00:00:00.000000Z"
    />
     by 
    bob
  </p>
</Info__ReferenceItemInfo>
`;
